### ISSUE TITLE
azure storage file download never returns to command prompt

### ISSUE THREAD

**Issue by cljung** [Azure/azure-xplat-cli#3212]
CLI Version: 0.10.5
OS Type: Mac / Win 
Installation via:  win-or-mac installer

Mode: ASM

Environment: AzureCloud

Description:
`azure storage file download [-q] sharename filename.ext local-path` command downloads file ok but never returns to command/shell prompt

Steps to reproduce:
1) set envvars AZUTE_STORAGE_ACCOUNT/ACCESS_KEY
2) run command "azure storage file download -q sharename filename.ext local-path"
see output ending with
info:  File saved as blabla
info:  storage file download command OK

Then it never returns to cmd prompt. No azure.err, etc. If you Ctrl+C you can continue to work, but BAT-files and bash-scripts hangs on this

**Comment by amarzavery** [comment]
@hasonmsft - Can you take a look at this issue ?

**Comment by cljung** [comment]
I'm a MSFTE, so you can reach me here also

Get Outlook for iOS




On Tue, Oct 18, 2016 at 12:03 AM +0200, "Amar Zavery" <notifications@github.com> wrote:





@hasonmsft - Can you take a look at this issue ?

**Comment by hasonmsft** [comment]
@cljung , use `--json` flag to have a try?

**Comment by cljung** [comment]
it completes when adding the --json flag, but should that be considered a bug or feature? azure file upload do not show the same behaviour

**Comment by hasonmsft** [comment]
@cljung , yes, it's a bug. We already filed a bug in our backlog to track it and will get you updated when it's fixed. :)

**Comment by cljung** [comment]
@hasonmsft - Thanks!

**Comment by shpros** [comment]
@hasonmsft, We're seeing this too. I've tried with 0.10.5 and 0.10.7, with nodejs 6.0.0 and 7.2.0. Running on 64-bit Windows 7 under git-bash. The `--json` trick unfortunately doesn't work for us. I've tried with and without `--json`, `-v`, `-vv`, and `-q`, but it's hanging on all combinations. FWIW, `azure storage file list` and `azure storage file show` do not have this hanging problem.

My command line looks like:

```
AZURE_STORAGE_ACCOUNT="${AZURE_STORAGE_ACCOUNT}" \
  AZURE_STORAGE_ACCESS_KEY="${AZURE_STORAGE_ACCESS_KEY}" \
  azure storage file download --share myshare -p myfile --json
```

Is there a link to the backlog bug in https://github.com/Azure/azure-storage-node/issues so I can watch it? I couldn't find one that looked related.

Thanks!

**Comment by hasonmsft** [comment]
@shpros , `--json` should work around this issue. We already fixed it and the PR get merged. You'll get the fix in next release. 

Here is the commit https://github.com/Azure/azure-xplat-cli/commit/89ed28f2616096f8659f837e2c86e31f58810c45 for your reference.
